### PR TITLE
Semantic filter

### PR DESCRIPTION
**Pull request by aravij**
Main file is `_semantic_filter.py`.
Semantic filtering use two following criteria:
 * There is no more than 1 declared variable in extraction opportunity, that is used later in method
 * `break` and `continue` statements may be only extracted with corresponding cycle

Added two flake warning to ignore list. Those warning was triggered if line break appear before or after logical operator.

Some refactoring done: tests are polished, word `cluster` changed to `extraction_opportunity`.

Unit tests added for semantic filtering. Added test for whole filtering for example from paper.

Added CLI for filtered opportunities.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -2,7 +2,7 @@
 universal = 1
 
 [flake8]
-ignore =
+ignore = W503, W504
```
Comment (lyriccoder) [inline setup.cfg:5]: why u r ignoring it? You should fix the problem, not to ignore it. 
@acheshkov Do u agree that we are not allowed to ignore them?

File: test/baselines/semi/utils.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        nonlocal extraction_opportunity_list
+        if isinstance(node, Statement) and node.node.line in statements_lines:
+            extraction_opportunity_list.append(node.node)
+
+    block_statement_graph.traverse(fill_extraction_opportunity)
+    return tuple(extraction_opportunity_list), block_statement_graph
+
+
+def get_method_ast(filename: str, class_name: str, method_name: str) -> AST:
+    current_directory = Path(__file__).absolute().parent
```
Comment (lyriccoder) [inline test/baselines/semi/utils.py:34]: what is that? You find parent directory, then again add the filename. Why do u need it?

**Review by lyriccoder — CHANGES_REQUESTED** [review]
need to discuss ignoring w503, w504

File: test/baselines/semi/utils.py (hunk 2 above)
Comment (aravij) [reply test/baselines/semi/utils.py:34]: Isn't a usual line we use in out unit tests? I just moved code, that extracts single java method into a separate function. Here I just store path to current directory. After I append given filename, which may differ from one to another method call.

File: veniq/baselines/semi/_common_types.py [hunk 3]
```diff
@@ -12,9 +13,30 @@ class StatementSemantic:
     used_methods: Set[str] = field(default_factory=set)
 
     def is_similar(self, other: "StatementSemantic") -> bool:
-        return len(self.used_objects & other.used_objects) != 0 or \
+        return len(self.used_objects_unwrapped & other.used_objects_unwrapped) != 0 or \
             len(self.used_methods & other.used_methods) != 0
```
Comment (acheshkov) [inline veniq/baselines/semi/_common_types.py:17]: do format it uniform

Comment (aravij) [reply veniq/baselines/semi/_common_types.py]: Formatted all files in PR.

**Review by lyriccoder — APPROVED** [review]
ok

**Pull request merged**